psi4/src/psi4/lib3index/dfhelper.cc

```diff
@@ -451,7 +457,24 @@ void DFHelper::prepare_AO() {
         count += size;
     }
 }
+void DFHelper::prepare_AO_wK() {
+    // prepare eris
+    std::shared_ptr<BasisSet> zero = BasisSet::zero_ao_basis_set();
+    auto rifactory = std::make_shared<IntegralFactory>(aux_, zero, primary_, primary_);
+    std::vector<std::shared_ptr<TwoBodyAOInt>> eri(nthreads_);
+#pragma omp parallel num_threads(nthreads_)
+    {
+        int rank = 0;
+#ifdef _OPENMP
+        rank = omp_get_thread_num();
+#endif
+        eri[rank] = std::shared_ptr<TwoBodyAOInt>(rifactory->eri());
+    }
 
+    // gather blocking info
+    std::vector<std::pair<size_t, size_t>> psteps;
+    std::pair<size_t, size_t> plargest = pshell_blocks_for_AO_build(memory_, 0, psteps);
```

Comment:
This function is incomplete as is and the technology required is identical to that of `prepare_AO`, but to make a new function which provide the compute helper for this. The type of function can be seen many times in `mintshelper.cc`.

Reply:
I think I get what you're saying.

The reason why some integrals get computed both in `prepare_AO_core()` and `prepare_AO_wK_core()` is that I saw something similar done in diskDFJK. I believed that that was done so that the same work would be done for the non-range separated components regardless of if `do_wK_` was set or not, but the exact reasoning for that decision rests with those who made it.

In any case, I've made a change along the lines of what you've suggested both in this function and in the upstream logic.